src/main/webapp/app/exam/manage/student-exams/student-exam-detail-table-row/student-exam-detail-table-row.component.ts

```diff
@@ -72,4 +70,31 @@ export class StudentExamDetailTableRowComponent implements OnChanges {
         }
         return route;
     }
+
+    getBonusPoints(exercise: Exercise): number | undefined {
+        if (!exercise) {
```

Comment:
Following the type the parameter cannot be `undefined`. Either this check is unnecessary, or the parameter type should be adapted.

Same for the other method below.

Additionally, a JSDoc for the method is missing.

Reply:
I adapted the `exercise` parameter type. The return type gives error when I try to remove `| undefined` because `exercise.maxPoints` and `exercise.bonusPoints` can be undefined.